modules/p2/ding_list/ding_list.module

```diff
@@ -0,0 +1,987 @@
+    global $user;
+    $account = $user;
+    
+    // Set the title depending on the list type.
+    $list = ding_list_get_list($view->args[0]);
+    $operations = ding_list_list_operations();
+    if (isset($operations[ding_base_get_value('ding_type', $list, 'field_list_type')]['_data'])) {
+      $data = $operations[ding_base_get_value('ding_type', $list, 'field_list_type')]['_data'];
+      
+      if (isset($data[$vars['title']])) {
+        $vars['title'] = $data[$vars['title']];
+      }
+      elseif (!empty($data['default_title'])) {
+        $vars['title'] = $data['default_title'];
+      }
+    }
+    
+    $dragging = ding_list_allowed($list, DING_LIST_OPERATION_SORT) && ding_list_user_has_access($list, 'edit', $account);
+    
+    // Enable dragging.
+    if ($dragging) {
+      $vars['theme_hook_suggestions'][] = 'ding-list--views-view-table.tpl.php';
+      $vars['attributes_array']['data-list-id'] = $list->id;
+      $table_id = drupal_html_id('ding-list');
+      $vars['attributes_array']['id'] = $table_id;
+      $vars['row_attributes'] = array();
+      foreach ($vars['result'] as $item) {
+        $vars['row_attributes'][]['data-item-id'] = $item->id;
+      }
+
+      $path = drupal_get_path('module', 'ding_list');
+      drupal_add_tabledrag($table_id, 'order', 'sibling', 'views-field-delta');
+      drupal_add_js($path . '/js/ding_list_tabledrag.js');
+      drupal_add_css($path . '/css/ding-list-tabledrag.css');
+      drupal_add_css($path . '/css/ding-list-view.css');
+    }
+    // Hide dragging fields if dragging is not enabled.
+    else {
+      if (isset($vars['field_classes']['id'])) {
+        foreach ($vars['field_classes']['id'] as &$classes) {
+          $classes .= ' element-hidden';
+        }
+      }
+      
+      if (isset($vars['field_classes']['delta'])) {
+        foreach ($vars['field_classes']['delta'] as &$classes) {
+          $classes .= ' element-hidden';
+        }
+      }
+    }
+  }
+}
+
+/**
+ * Change the html header for lists, to ensure og:url and og:description.
+ */
+function ding_list_html_head_alter(&$head_elements) {
+  // If the current page is a list.
+  if (arg(0) === 'list' && is_numeric(arg(1)) && !arg(2)) {
+    $list = ding_list_get_list(arg(1));
+    $head_elements['ogurl'] = array(
+      '#type' => 'html_tag',
+      '#tag' => 'meta',
+      '#attributes' => array(
+        'property' => 'og:url',
+        'content' => url('list/' . arg(1), array('absolute' => TRUE)),
+      ),
+    );
+    $head_elements['ogdescription'] = array(
+      '#type' => 'html_tag',
+      '#tag' => 'meta',
+      '#attributes' => array(
+        'property' => 'og:description',
+        'content' => ding_base_get_value('ding_type', $list, 'field_notes'),
+      ),
+    );
+  }
+}
+
+/**
+ * Implements hook_entity_load().
+ *
+ * When a list is loaded convert the cql of ting_objects to the new cql.
+ */
+function ding_list_entity_load($entities, $type) {
```

Comment:
If we can convert elements on load could we do it on install as well?

In that case the conversion would run once and for all. This way we add complexity to every entity_load.